app/assets/javascripts/modules/accordion-with-descriptions.js

```diff
@@ -188,71 +214,121 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
-          $node.attr("aria-expanded", "false");
-        } else {
-          $node.attr("aria-expanded", "true");
-        }
+      this.open = function () {
+        // Show the subsection content
+        that.$subsectionContent.removeClass('js-hidden');
+        // Swap the plus and minus sign
+        $subsectionElement.removeClass('subsection');
+        $subsectionElement.addClass('subsection-is-open');
+        // Tell impaired users that the section is open
+        that.$subsectionButton.attr("aria-expanded", "true");
       }
 
-      function openSection($node) {
-        $node.removeClass('js-hidden');
+      this.close = function () {
+        // Hide the subsection content
+        that.$subsectionContent.addClass('js-hidden');
+        // Swap the plus and minus sign
+        $subsectionElement.removeClass('subsection-is-open');
+        $subsectionElement.addClass('subsection');
+        // Tell impaired users that the section is closed
+        that.$subsectionButton.attr("aria-expanded", "false");
       }
 
-      function closeSection($node) {
-        $node.addClass('js-hidden');
+      this.isClosed = function () {
+        return that.$subsectionContent.hasClass('js-hidden');
       }
+    }
 
-      function showOpenIcon($node) {
-        $node.parent().removeClass('subsection');
-        $node.parent().addClass('subsection-is-open');
+    // A contructor for an object that represents a click event on a subsection which
+    // handles the complexity of sending different tracking labels to Google Analytics
+    // depending on which part of the subsection the user clicked.
+    function SubsectionToggleClick (subsectionView, event) {
+      var that = this;
+
+      this.$target = $(event.target);
+
+      this.track = function () {
+        track('pageElementInteraction', that._trackingAction(), { label: that._trackingLabel() });
+      }
+
+      this._trackingAction = function () {
+        return (subsectionView.isClosed() ? 'accordionClosed' : 'accordionOpened');
+      }
+
+      this._trackingLabel = function () {
+        if (that._clickedOnIcon()) {
+          return subsectionView.title + ' - ' + that._iconType() + ' Click';
+        } else if (that._clickedOnHeading()) {
+          return subsectionView.title + ' - Heading Click';
+        } else {
+          return subsectionView.title + ' - Click Elsewhere';
+        }
+      }
+
+      this._clickedOnIcon = function () {
+        return that.$target.hasClass('subsection-icon');
       }
 
-      function showCloseIcon($node) {
-        $node.parent().removeClass('subsection-is-open');
-        $node.parent().addClass('subsection');
+      this._clickedOnHeading = function () {
+        return that.$target.hasClass('js-subsection-button');
       }
 
-      function setExpandedState($node, state) {
-        $node.attr("aria-expanded", state);
+      this._iconType = function () {
+        return (subsectionView.isClosed() ? 'Minus' : 'Plus');
```

Comment:
(This is potentially susceptible to a race condition, but again I recognise that this is a read-across)